### PR TITLE
Django 1.8 support

### PR DESCRIPTION
**Pull request by pzrq**
Django 1.8 beta 1 was released recently.
https://www.djangoproject.com/weblog/2015/feb/25/releases/

This branch makes a start on supporting Django 1.8 on top of the `next` branch, though I don't know how to get past the last failing test (edit: when running `tox`):

ERROR: test_29_medialibrary_admin (testapp.tests.test_page.PagesTestCase)
...
AttributeError: 'ManyToManyRel' object has no attribute 'parent_model'

### CODE REVIEW

**Comment by matthiask** [comment]
Thanks! I'll test this in a project of mine.

**Comment by pzrq** [comment]
Thanks @matthiask, feel free to cherry-pick bits and pieces of the combined branch as needed - this ended up as quite a few changes and we didn't need the ManyToManyRel change to test it worked in our project without issues AFAIK.

File: setup.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ def read(filename):
         'django-mptt>=0.6.0',
```
Comment (nanuxbe) [inline setup.py]: django-mptt 0.7.1 has been released a few days ago. If I understand it well, this was blocking for this PR.

**Comment by matthiask** [comment]
Thank you! Not sure why the code in `feincms/__init__.py` would not be necessary anymore. Care to comment? 

Merged.

**Comment by DrMeers** [comment]
@matthiask did you see this https://github.com/django/django/commit/fb48eb058 ?

**Comment by matthiask** [comment]
@DrMeers Yes, thank you. I still have the suspicion that clearing caches is still necessary in one way or the other, if only because we spent to much time hunting problems down in this area. The meta API formalization certainly helps here.

**Comment by DrMeers** [comment]
from my quick skim of the code a few days ago it looked like the field data was no longer cached, but I may have missed something

**Comment by matthiask** [comment]
Inside `django/db/models/options.py` I see:

```
    @cached_property
    def related_objects(self):
        """
```

That's not the only cached property I'm seeing. 

The issue was #323. Not a data loss issue, but degraded / broken functionality. Maybe we should just leave the code as it is now and see whether we get any problem reports.

**Comment by DrMeers** [comment]
I clearly didn't skim exhaustively :)

I wonder if we may just need to use `model._meta._expire_cache()` instead of our `model._meta._fill_fields_cache()`...

**Comment by pzrq** [comment]
Thanks for merging @matthiask, looking forward to future releases.

**Pull request closed without merging**